### PR TITLE
Use SQLAlchemy to preload the moderation information instead of the service

### PR DESCRIPTION
**Pull request by jon-betts**
For: https://github.com/hypothesis/h/issues/6769

This needs a good few things to go through first before it can be sensibly worked on

### CODE REVIEW

File: h/formatters/annotation_hidden.py [hunk 1]
```diff
@@ -18,36 +18,23 @@ class AnnotationHiddenFormatter:
...
 
-        if self._is_hidden(annotation):
-            return {"hidden": True, "text": "", "tags": []}
+        result = {"hidden": True}
 
-        return {"hidden": False}
+        if not self._current_user_is_moderator(annotation):
+            result.update({"text": "", "tags": []})
+
+        return result
```
Comment (jon-betts) [inline h/formatters/annotation_hidden.py:37]: I've rearranged this code after writing the tests in a way I think actually reveals the intent:

 * We return hidden false if it's not hidden, or you are the author
 * From here on we say it's hidden
 * For non-moderators we censor the content

File: tests/h/services/annotation_json_presentation_test/service_test.py [hunk 2]
```diff
@@ -53,6 +54,40 @@ def test_present_all(self, svc, factories, annotation, AnnotationJSONPresenter):
...
+            def __call__(self, *args, **kwargs):
+                print(args, kwargs)
+                self.count += 1
+
+            def reset(self):
+                self.count = 0
+
+        query_counter = QueryCounter()
+        event.listen(db_engine, "before_cursor_execute", query_counter)
+        return query_counter
```
Comment (jon-betts) [inline tests/h/services/annotation_json_presentation_test/service_test.py:89]: This was a new test which covered both the new and old preloading stuff. Kind of pain to setup, but proves the point

File: package-lock.json [hunk 3]
```diff
@@ -1,12460 +1,8 @@
 {
   "name": "h",
   "version": "0.0.0",
-  "lockfileVersion": 2,
+  "lockfileVersion": 1,
   "requires": true,
-  "packages": {
```
Comment (marcospri) [inline package-lock.json:6]: Not sure what'ds going on here, you might have taken a shortcut to remove lots of lines?

Comment (jon-betts) [reply package-lock.json]: Eugh. This is a really annoying file that gets regenerated all the time and makes a mess of the checkout status. So I've committed it by mistake. I've fixed this now.

File: tests/h/services/annotation_json_presentation_test/service_test.py [hunk 4]
```diff
@@ -53,6 +54,39 @@ def test_present_all(self, svc, factories, annotation, AnnotationJSONPresenter):
...
+        db_session.flush()
+        db_session.expire(annotation)
+        if with_preload:
+            svc.present_all([annotation.id])
+
+        query_counter.reset()
+        getattr(annotation, property)
+
+        # If we preloaded, we shouldn't execute any queries (and vice versa)
+        assert bool(query_counter.count) != with_preload
```
Comment (marcospri) [inline tests/h/services/annotation_json_presentation_test/service_test.py:72]: :+1:

**Review by marcospri — APPROVED** [review]

**Pull request merged**